### ISSUE TITLE
subclip: runtime error when audio frame is completely missing

### ISSUE THREAD

**Issue by aliyilmazz** [Zulko/moviepy#772]
hello, I'm implementing a video splitter microservice using moviepy.

One of my test videos caused a runtime error. I thought you might want to know about it.

Recall `subclip(clip_start, clip_end)`: if audio frames are missing somewhere between `clip_start` and `clip_end`, moviepy automatically duplicates last seen audio frame to following frames. See pull request: https://github.com/Zulko/moviepy/pull/377

However, given a subclip, if there is NO audio frame at all, this pull request is not capable of handling this case and library gives runtime error:
```
IndexError: index -1 is out of bounds for axis 0 with size 0
```

Example use-case: I have a 27 second-long video, which has no audio frames after t=25.
if I try to subclip(15,27), that's fine. After t=25, audio repeats itself.
if I try to subclip(26,27), thats where I get runtime error.

For this issue, I came up with a solution like this:
```python
    clip = VideoFileClip(tempvideopath).subclip(clip_start, clip_end)
    try:
        clip.write_videofile(tempfilename, codec=task['codec'])
    except IndexError:
        try:
            clip.write_videofile(tempfilename, codec=task['codec'], audio=False)
        except Exception as e:
            logger.info("exception caught: %s" % e)
```

Basically, once I get an `IndexError` exception, I try to render it without audio. 

I can put it into library and send a pull request, if you wish. In any case, I thought you might wanna know about this.

Cheers,
Ali

**Comment by tburrows13** [comment]
Thanks for spotting this. Feel free to create a PR for it if you like.

**Comment by aliyilmazz** [comment]
Alright, I surely will contribute in my spare time and mention it here, hopefully within this week. Cheers

**Comment by keikoro** [comment]
@aliyilmazz Any update on this?

**Comment by racket99** [comment]
I have the same issue

**Comment by tburrows13** [comment]
I think I'd prefer a better solution than try/except, perhaps someone would be able to investigate?

**Comment by keikoro** [comment]
Closing this issue due to lack of response from OP in several years.
